pkg/result/filter.go

```diff
@@ -69,8 +72,16 @@ func FilterResult(ctx context.Context, result *types.Result, ignoreConf IgnoreCo
 	filterLicenses(result, severities, opt.IgnoreLicenses, ignoreConf)
 
 	if opt.PolicyFile != "" {
-		if err := applyPolicy(ctx, result, opt.PolicyFile); err != nil {
-			return xerrors.Errorf("failed to apply the policy: %w", err)
+		// Get ignore policy files from the input path (either file or files in dir)
+		policyFiles, err := findPolicyFiles(opt.PolicyFile)
+		if err != nil {
+			return err
+		}
+
+		for _, policyFile := range policyFiles {
```

Comment:
IMHO, loading all Rego files together and evaluating the policies collectively is more intuitive for Rego users than loading and repeatedly evaluating individual Rego files. The current implementation may not work correctly if variable definitions are spread across multiple files.

To address this, using the `rego.Load()` function to load all Rego files from a directory recursively would be more appropriate. This ensures that all files are considered together, allowing for proper resolution of variables, rules, and dependencies.

However, since I've been away from OPA recently, I would greatly appreciate insights from @simar7 to ensure the correctness and effectiveness of this approach.

Reply:
+1 I'd also like to mention that running `.PrepareForEval()` is an expensive operation. Doing it recursively on a directory that is user defined to load all files that match the Rego extension, whether relevant or not, can be costly. 

However if we go down the route of loading all rego files via `rego.Load()` as @knqyf263 mentioned, I'm not sure if we can have multiple checks that can contain the same default as the compiler will error out. Since Rego checks often contain defaults that result in "fail-close" type of checks, this would be a common occurrence (e.g.  multiple checks having `default allow=false`).

Maybe a safer route is to allow this flag to have values that can be a list of filenames rather than a directory? This would limit the scope of rego files that get loaded.

Reply:
Okay, we need to understand the use case precisely. @dstrelbytskyi Could you elaborate on it?

Reply:
@knqyf263 I want to come back to this and elaborate.

So the idea is to be able to provide multiple policy files for filtering the scan results. That's not about files that depend on each other and share any variables (and thus need to be loaded together). Every file is standalone and may have an own default definition. Every file is just applied sequentially for the result filtering.

I like the idea of taking the list of filenames instead of recursively scanning the directory. Say the `PolicyFile` option may take (as an addition to a current single file path) a comma separated list of files. That completely makes sense to me and will work for sure.

About the other concern. Above, we learned from **simar7** that the policy evaluation preparation (`PrepareForEval`) is an expensive operation. In the current filtering implementation (with a single provided file) it does load and prepare the policy file **for each result** being filtered and that indeed looks inefficient. What if the policy file (or multiple files) is loaded and prepared once, then the `PreparedEvalQuery` object for the policy file is cached and then reused for the further results evaluation? Does that make sense?

How does that sound for you? I can make the mentioned changes an open a new PR (since this is already stale).

Reply:
@knqyf263 sorry for tagging one more time. Just to make sure the message above isn't missed.